components/nexus-bootstrap/src/main/java/org/sonatype/nexus/bootstrap/jetty/ConnectorManager.java

```diff
@@ -0,0 +1,262 @@
+      throw new UnsupportedHttpSchemeException(httpScheme);
+    }
+    serverConnector.setHost(connectorPrototype.getHost());
+    serverConnector.setPort(connectorConfiguration.getPort());
+    serverConnector.setIdleTimeout(connectorPrototype.getIdleTimeout());
+    serverConnector.setSoLingerTime(connectorPrototype.getSoLingerTime());
+    serverConnector.setAcceptorPriorityDelta(connectorPrototype.getAcceptorPriorityDelta());
+    serverConnector.setSelectorPriorityDelta(connectorPrototype.getSelectorPriorityDelta());
+    serverConnector.setAcceptQueueSize(connectorPrototype.getAcceptQueueSize());
+
+    managedConnectors.put(connectorConfiguration, serverConnector);
+    server.addConnector(serverConnector);
+    try {
+      serverConnector.start();
+    }
+    catch (Exception e) {
+      throw new RuntimeException(e); // NOSONAR: no guava in scope
+    }
+
+    return serverConnector;
+  }
+
+  /**
+   * Stops and removes the connector configuration from Jetty.
+   */
+  public void removeConnector(final ConnectorConfiguration connectorConfiguration) {
+    final ServerConnector serverConnector = managedConnectors.remove(connectorConfiguration);
+    if (serverConnector != null) {
+      try {
+        serverConnector.stop();
+      }
+      catch (Exception e) {
+        throw new RuntimeException(e); // NOSONAR: no guava in scope
+      }
+      server.removeConnector(serverConnector);
+    }
+  }
+
+  // ==
+
+  /**
+   * Verifies all the needed bits are present in Jetty XML configuration (as HTTPS must be enabled by users).
+   */
+  private void verifyConfiguration(final HttpScheme httpScheme) {
+    try {
+      if (HttpScheme.HTTP == httpScheme) {
+        bean(HTTP_CONFIG_ID, HttpConfiguration.class);
+        bean(HTTP_CONNECTOR_ID, ServerConnector.class);
+      }
+      else if (HttpScheme.HTTPS == httpScheme) {
+        bean(SSL_CONTEXT_FACTORY_ID, SslContextFactory.class);
+        bean(HTTPS_CONFIG_ID, HttpConfiguration.class);
+        bean(HTTPS_CONNECTOR_ID, ServerConnector.class);
+      }
+      else {
+        throw new UnsupportedHttpSchemeException(httpScheme);
+      }
+    }
+    catch (IllegalStateException e) {
+      throw new IllegalStateException("Jetty HTTPS is not enabled in Nexus", e);
+    }
+  }
+
+  private List<ConnectorConfiguration> buildDefaultConnectors() {
+    final List<ConnectorConfiguration> result = new ArrayList<>();
+    try {
+      verifyConfiguration(HttpScheme.HTTP);
+      final int port = defaultConnector(HttpScheme.HTTP).getPort();
+      result.add(new ConnectorConfiguration()
+      {
+        @Override
+        public HttpScheme getScheme() {
+          return HttpScheme.HTTP;
+        }
+
+        @Override
+        public int getPort() {
+          return port;
+        }
+
+        @Override
+        public HttpConfiguration customize(final HttpConfiguration configuration) {
+          return configuration;
+        }
+      });
+    }
+    catch (IllegalStateException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)